pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -348,18 +388,60 @@ class RpmContentSaver(ContentSaver):
         """
         Save a batch of UpdateCollection, UpdateCollectionPackage, UpdateReference objects.
 
+        When it has a treeinfo file, save a batch of Addon, Checksum, Image, Variant objects.
+
         Args:
             batch (list of :class:`~pulpcore.plugin.stages.DeclarativeContent`): The batch of
                 :class:`~pulpcore.plugin.stages.DeclarativeContent` objects to be saved.
 
         """
+        def _handle_distribution_tree(declarative_content):
+            distribution_tree = declarative_content.content
+            kickstart_data = declarative_content.extra_data
+            repository_id = kickstart_data["repository_id"]
+
+            addons = []
+            checksums = []
+            images = []
+            variants = []
+
+            for addon in kickstart_data["addons"]:
```

Comment:
We need to actually download and save the artifacts backing this model. I think that is probably best done in another stage so that it occurs outside of this batche's transaction in this stage.